creator-node/src/snapbackSM.js

```diff
@@ -47,63 +61,387 @@ class SnapbackSM {
+            if (user.secondary2) secondary2 = user.secondary2
+            let primaryClockValue = primaryClockValues[userWallet]
+            let secondary1ClockValue = secondary1 != null ? secondaryNodeUserClockStatus[secondary1][userWallet] : undefined
+            let secondary2ClockValue = secondary2 != null ? secondaryNodeUserClockStatus[secondary2][userWallet] : undefined
+            let secondary1SyncRequired = secondary1ClockValue === undefined ? true : primaryClockValue > secondary1ClockValue
+            let secondary2SyncRequired = secondary2ClockValue === undefined ? true : primaryClockValue > secondary2ClockValue
+            this.log(`${userWallet} primaryClock=${primaryClockValue}, (secondary1=${secondary1}, clock=${secondary1ClockValue} syncRequired=${secondary1SyncRequired}), (secondary2=${secondary2}, clock=${secondary2ClockValue}, syncRequired=${secondary2SyncRequired})`)
+            // Enqueue sync for secondary1 if required
+            if (secondary1SyncRequired && secondary1 != null) {
+              await this.issueSecondarySync(userWallet, secondary1, this.endpoint)
+              numSyncsIssued += 1
+            }
+            // Enqueue sync for secondary2 if required
+            if (secondary2SyncRequired && secondary2 != null) {
+              await this.issueSecondarySync(userWallet, secondary2, this.endpoint)
+              numSyncsIssued += 1
+            }
+          } catch (e) {
+            this.log(`Caught error for user ${user.wallet}, ${JSON.stringify(user)}, ${e.message}`)
+          }
+        }
+      )
+    )
+
+    let previousModuloSlice = this.currentModuloSlice
+
+    // Increment and adjust current slice by ModuloBase
+    this.currentModuloSlice += 1
+    this.currentModuloSlice = this.currentModuloSlice % ModuloBase
+
+    this.log(`Updated modulo slice from ${previousModuloSlice} to ${this.currentModuloSlice}`)
+    this.log(`Issued ${numSyncsIssued} sync ops`)
+    this.log(`------------------END Process SnapbackSM Operation, slice ${previousModuloSlice} ------------------`)
+  }
+
+  // Track an ongoing sync operation to a secondary
+  async monitorSecondarySync (syncWallet, primaryClockValue, secondaryUrl) {
+    let startTime = Date.now()
+    // Monitor the sync status
+    let syncMonitoringRequestParameters = {
+      method: 'get',
+      baseURL: secondaryUrl,
+      url: `/sync_status/${syncWallet}`,
+      responseType: 'json'
+    }
+    let syncAttemptCompleted = false
+    while (!syncAttemptCompleted) {
       try {
-        await this.processStateMachineOperation(job)
+        let syncMonitoringResp = await axios(syncMonitoringRequestParameters)
+        let respData = syncMonitoringResp.data.data
+        this.log(`processSync ${syncWallet} secondary response: ${JSON.stringify(respData)}`)
+        // A success response does not necessarily guarantee completion, validate response data to confirm
+        if (respData.clockValue === primaryClockValue) {
```

Comment:
thoughts on changing this to be >=? only edge case i can think of is
1. you fetch the clock values and store in memory
2. before you trigger an update, user does an action and it's confirmed by primary and replicas
3. this check will fail because clock would have changed for both nodes and it'll be forced to time out

Reply:
yes i agree with this fully